### PR TITLE
Improve the logic of comparing the scores of two players.

### PR DESCRIPTION
**Pull request by iannbing**
Currently the logic of comparing scores is incomplete: didn't consider a tie in the hand cards.
This PR fixes this issue and adds unit tests for these cases.

### CODE REVIEW

File: love_letter/models/__init__.py [hunk 1]
```diff
@@ -270,8 +269,14 @@ def __repr__(self):
         return f"Player({self.name},{self.cards})"
 
     def __gt__(self, other: "Player"):
+        # Players are comparable only if each player has only 1 card.
         if len(self.cards) == 1 and len(other.cards) == 1:
-            return self.cards[0].value > other.cards[0].value
+            return (
+                self.cards[0].value > other.cards[0].value
+                or self.total_value_of_card > other.total_value_of_card
+            )
```
Comment (s9891326) [inline love_letter/models/__init__.py:277]: 這邊的邏輯有點問題XD，像是這個案例的話
```
player_1 = [4, 8]
player_2 = [5, 3]

player_1[0] > player_2[0] or player_1[1] > player_2[1]  # True => False才對
```

File: love_letter/models/__init__.py [hunk 2]
```diff
@@ -270,8 +269,13 @@ def __repr__(self):
...
     def __gt__(self, other: "Player"):
-        if len(self.cards) == 1 and len(other.cards) == 1:
-            return self.cards[0].value > other.cards[0].value
+        if len(self.cards) != 1 or len(other.cards) != 1:
+            raise AssertionError("Unable to compare players.")
+        if self.cards[0].value > other.cards[0].value:
+            return True
+        if self.cards[0].value == other.cards[0].value:
+            return self.total_value_of_card > other.total_value_of_card
+        return False
```
Comment (IamSaker) [inline love_letter/models/__init__.py:278]: Nitpick: for reference only
```python
if len(self.cards) != 1 or len(other.cards) != 1:
    raise AssertionError("Unable to compare players.")
elif self.cards[0].value > other.cards[0].value:
    return True
elif self.cards[0].value == other.cards[0].value:
    return self.total_value_of_card > other.total_value_of_card
else:
    return False
````

Comment (IamSaker) [reply love_letter/models/__init__.py:278]: simplify
```python
if len(self.cards) != 1 or len(other.cards) != 1:
    raise AssertionError("Unable to compare players.")
elif self.cards[0].value == other.cards[0].value:
    return self.total_value_of_card > other.total_value_of_card
else:
    return self.cards[0].value > other.cards[0].value
````

Comment (iannbing) [reply love_letter/models/__init__.py]: @IamSaker  Good one! Thanks! Pushed the changes.

**Pull request merged**